pkg/proxy/topology.go

```diff
@@ -43,11 +64,15 @@ func FilterTopologyEndpoint(currentNodeName types.NodeName, nodeMap NodeMap, top
 			if nodeName == "" {
 				continue
 			}
-			node, ok := nodeMap[nodeName]
-			if !ok {
+			node, found := nodeMap[nodeName]
+			if !found {
+				continue
```

Comment:
Above (L45), for the current node, if we do not find the node in the map, then we just return the endpoints (as if this feature didn't exist), effectively *including all endpoints*. Now, in here, as we examine each individual endpoint, if we don't find the node in the map, we *exclude* the endpoint. Is that what we want? Is that the "safest" behavior? Given that the topology is intended as an efficiency feature not a security feature, wouldn't it make more sense to *include* the endpoint here by default, to ensure availability?

Reply:
I think that if you ask for topology and we can't determine a node's topology we should not include it.  BUt that's a first-cut feeling.  Concretely, it would be wrong to use another node for same-node services.  It might even be a violation of regulation or data sovereignty to get this wrong.

Reply:
Agree with Tim, we should assume there is no endpoint at first, then include endpoints which match the topology.

Reply:
Ok. In that case, if we cannot determine the topology of the current node, shouldn't we also not have any endpoints? Otherwise we could violate the constraint. Also, if they user has asked for "", we should probably include this even if we don't know the topology.

Granted, this is a very small edge case. But the logic is:

- if "" is not in the keys, then do not add any endpoints if we cannot determine this node topology, and do not add any endpoints for which we cannot determine topology
 - if "" is included in the keys, then include an endpoint even if we cannot determine this node's or the endpoint node's topology

Reply:
Oops, sorry for almost missed your comments. Nice catch! edge cases are considered in the new commit:
- include all endpoints if topology key not empty and "Any" specified when we cannot determine current node's topology
- do not include any endpoints if topology key not empty and "Any" is not specified when we cannot determine current node's topology